erpnext/regional/india/e_invoice/utils.py

```diff
@@ -885,7 +885,10 @@ def log_request(self, url, headers, data, res):
 			}
 		)
 		request_log.save(ignore_permissions=True)
-		frappe.db.commit()
+
+		if (self.invoice.doctype, self.invoice.name) not in frappe.flags.currently_saving:
```

Comment:
Yes, I was aware of this. 
The problem is we can't run `frappe.db.commit` as it will commit all the previous changes to the `db` and lead to partial submissions in case

If IRN is to be generated on submission and submission is just one single process, then we'll have to prevent `commit` unless the submission is successful

Do you have any solution for this?

Reply:
I am not sure but can't we use something like doc.get_doc_before_save() or doc.reload() to revert all changes before we commit?